File: src/node/magma/app/BooleanType.ts
import { Type } from "../../magma/api/Type";
import { Platform } from "../../magma/app/Platform";
class BooleanType implements Type {
	platform: Platform;
	constructor (platform: Platform) {
		this.platform = platform;
	}
	generate(): string {
		if (Platform.TypeScript === this.platform) {
			return "boolean";
		}
		return "Bool";
	}
	isFunctional(): boolean {
		return false;
	}
	isVar(): boolean {
		return false;
	}
	generateBeforeName(): string {
		return "";
	}
}

File: src/java/magma/app/BooleanType.java
package magma.app;

import magma.api.Type;

record BooleanType(Platform platform) implements Type {
    @Override
    public String generate() {
        if (Platform.TypeScript == this.platform) {
            return "boolean";
        }

        return "Bool";
    }

    @Override
    public boolean isFunctional() {
        return false;
    }

    @Override
    public boolean isVar() {
        return false;
    }

    @Override
    public String generateBeforeName() {
        return "";
    }
}
